pydantic/main.py

```diff
@@ -206,12 +206,14 @@ class BaseModel(metaclass=MetaModel):
         _schema_cache: Dict[Any, Any] = {}
 
     Config = BaseConfig
-    __slots__ = ('__values__',)
+    __slots__ = ('__values__', '__fields_set__')
 
     def __init__(self, **data: Any) -> None:
         if TYPE_CHECKING:  # pragma: no cover
             self.__values__: Dict[str, Any] = {}
-        self.__setstate__(self._process_values(data))
+            self.__fields_set__: Set[str] = set()
+        fields_set = data.pop("__fields_set__", set(data.keys()))
```

Comment:
Strict rule in pydantic that you can;t pass anything other than values to the class like this.

Users should be able to create a model with `MyModel(**whatever)` without having to think about what's in `whatever`.

Remove this and either call `__setstate__` directly, or set `model.__fields_set__ = thing` directly.